pkg/server/api/bundle/v1/service.go

```diff
@@ -175,7 +188,36 @@ func parseJWTAuthorities(keys []*types.JWTKey) ([]*common.PublicKey, error) {
 }
 
 func (s *Service) PublishJWTAuthority(ctx context.Context, req *bundle.PublishJWTAuthorityRequest) (*bundle.PublishJWTAuthorityResponse, error) {
-	return nil, status.Errorf(codes.Unimplemented, "method PublishJWTAuthority not implemented")
+	log := rpccontext.Logger(ctx)
+
+	if err := rpccontext.RateLimit(ctx, 1); err != nil {
+		log.WithError(err).Error("Rejecting request due to key publishing rate limiting")
+		return nil, err
+	}
+
+	if req.JwtAuthority == nil {
+		log.Error("Invalid request: missing JWT authority")
+		return nil, status.Error(codes.InvalidArgument, "missing JWT authority")
+	}
+
+	keys, err := parseJWTAuthorities([]*types.JWTKey{req.JwtAuthority})
+	if err != nil {
+		log.WithError(err).Error("Invalid request: invalid JWT authority")
+		return nil, status.Errorf(codes.InvalidArgument, "invalid JWT authority: %v", err)
+	}
+
+	resp, err := s.up.PublishJWTKey(ctx, keys[0])
+	if err != nil {
+		log.WithError(err).Error("Failed to publish JWT key")
+		return nil, status.Errorf(codes.Internal, "failed to publish JWT key: %v", err)
+	}
+
+	// TODO: after pushing authoriry, may we reset dsCache?
```

Comment:
Looks like we are not implementing any caching like we currently have in the Node API handler. I guess the question is if we want to have some caching in this new implementation? @azdagron Thoughts?

Reply:
I think the question of caching is a little more complicated now that all of the calls that manipulated the cache are no longer in a single service handler. We need to think this over. I think we're ok not trying to introduce caching in this PR.

Reply:
I also think that we don't need to implement caching in this PR. I filed #1631 to explore the implementation of a datastore cache, so we can track this.

Reply:
I'll rebase and add that issue number to TODO